commit fcd208f71b3319044829ef1b384bf2c7a28b449b
Author: Jens Axboe <jaxboe@fusionio.com>
Date:   Fri Oct 29 08:10:18 2010 -0600

    block: check for proper length of iov entries in blk_rq_map_user_iov()
    
    Ensure that we pass down properly validated iov segments before
    calling into the mapping or copy functions.
    
    Reported-by: Dan Rosenberg <drosenberg@vsecurity.com>
    Cc: stable@kernel.org
    Signed-off-by: Jens Axboe <jaxboe@fusionio.com>

diff --git a/block/blk-map.c b/block/blk-map.c
index 0b1af5a..71e102e 100644
--- a/block/blk-map.c
+++ b/block/blk-map.c
@@ -191,6 +191,8 @@ int blk_rq_map_user_iov(struct request_queue *q, struct request *rq,
 			unaligned = 1;
 			break;
 		}
+		if (!iov[i].iov_len)
+			return -EINVAL;
 	}
 
 	if (unaligned || (q->dma_pad_mask & len))
